snap_core22/snapcraft.yaml

```diff
@@ -0,0 +1,270 @@
+name: plotjuggler
+adopt-info: plotjuggler # parse metadata from the plotjuggler part
+summary: The timeseries visualization tool that you deserve
+description: |
+  QT5 based application to display time series in plots,
+  using an intuitive "drag and drop" interface.
+
+  The snap comes with only ROS 2 plugins.
+  You can launch it with
+
+    $ plotjuggler
+
+  launching PlotJuggler with ROS 2 plugins by default.
+
+issues: https://github.com/facontidavide/plotjuggler/issues
+source-code: https://github.com/facontidavide/plotjuggler
+license: MPL-2.0
+
+confinement: strict
+base: core22
+
+apps:
+  plotjuggler:
+    command: usr/bin/launcher-plotjuggler-ros2
+    plugs: [network, network-bind, home, removable-media]
+    extensions: [kde-neon, ros2-humble]
+
+parts:
+  plotjuggler:
+    plugin: cmake
+    source: .
+    cmake-parameters:
+      - -DCMAKE_BUILD_TYPE=Release
+      - "-DCMAKE_PREFIX_PATH=$(echo $SNAPCRAFT_CMAKE_ARGS | awk -F= '{printf(\"%s/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/cmake/Qt5\", $2)}')"
+    build-packages:
+      - distro-info-data
+      - libpulse0
+      - libdw-dev
+      - libbfd-dev
+      - libdwarf-dev
+      - libprotoc-dev
+      - libgl-dev
+      - libmosquitto-dev
+      - libzmq3-dev
+      - libzstd-dev
+    stage-packages:
+      - libdw1
+      - libmosquitto1
+      - libprotobuf23
+      - libprotobuf-dev
+      - libzmq5
+      - libzstd1
+      - libfreetype6
+    override-pull: |
+        craftctl default
+
+        version="$(git describe --always --tags| sed -e 's/^v//;s/-/+git/;y/-/./')"
+        [ -n "$(echo $version | grep "+git")" ] && grade=devel || grade=stable
+        snapcraftctl set-version "$version"
+        snapcraftctl set-grade "$grade"
```

Comment:
`craftctl set version` could be used here from what [this guide](https://forum.snapcraft.io/t/micro-howto-migrate-from-core20-to-core22/30188) says.
Sme for the grade.